DcNotificationService/NotificationService.swift

```diff
@@ -14,21 +14,36 @@ class NotificationService: UNNotificationServiceExtension {
             contentHandler(silenceNotification())
             return
         }
-        UserDefaults.setNseFetching()
+        UserDefaults.setNseFetching(until: Date().addingTimeInterval(26))
 
         // as we're mixing in notifications from accounts without PUSH and we cannot add multiple notifications,
         // it is best to move everything to the same thread - and set just no threadIdentifier
 
         dcAccounts.openDatabase(writeable: false)
         let eventEmitter = dcAccounts.getEventEmitter()
 
-        if !dcAccounts.backgroundFetch(timeout: 25) {
-            UserDefaults.pushToDebugArray("ERR3")
-            UserDefaults.setNseFetching(false)
+        // Send the bestAttempt notification when memory is critical because the process will be killed
+        // by the system soon and any notification is better than nothing.
+        var exitedDueToCriticalMemory = false
+        let memoryPressureSource = DispatchSource.makeMemoryPressureSource(eventMask: .critical)
+        memoryPressureSource.setEventHandler {
+            // Order of importance because we might crash very soon
+            // TODO: Could notify with "received message too big to decrypt in background"
```

Comment:
usually, we avoid "TODO" in the source code, this usually stays forever. in this case, i also think "New Messages" is totally fine, there are also some other situations where this may happen

```suggestion
```